src/ucp/core/ucp_worker.h

```diff
@@ -275,7 +286,10 @@ typedef struct ucp_worker {
     unsigned                         num_active_ifaces;   /* Number of activated ifaces  */
     ucp_tl_bitmap_t                  scalable_tl_bitmap;  /* Map of scalable tl resources */
     ucp_worker_cm_t                  *cms;                /* Array of CMs, one for each component */
-    ucs_mpool_t                      am_mp;               /* Memory pool for AM receives */
+    uint32_t                         am_mps_bitmap;       /* Bitmap of AM receive mpool sizes */
+    ucs_mpool_t                      *am_mps;             /* Memory pools for AM receives */
+    /* Map of AM mpools by log of size */
+    ucs_mpool_t                      *am_mps_map[UCP_WORKER_AM_MPS_MAP_SIZE];
```

Comment:
can we keep only am_mps_map and not am_mps?
am_mps_bitmap should tell which are valid